backend/experiment/rules/tests/test_congosamediff.py

```diff
@@ -0,0 +1,115 @@
+        self.participant = Participant.objects.create()
+        self.experiment = Experiment.objects.create(
+            name='CongoSameDiff',
+            slug='congosamediff',
+            rounds=4,
+        )
+        self.session = Session.objects.create(
+            experiment=self.experiment,
+            participant=self.participant,
+            playlist=self.playlist
+        )
+
+    def test_initializes_correctly(self):
+        congo_same_diff = CongoSameDiff()
+        assert congo_same_diff.ID == 'CONGOSAMEDIFF'
+        assert congo_same_diff.contact_email == 'aml.tunetwins@gmail.com'
+
+    def test_first_round(self):
+        congo_same_diff = CongoSameDiff()
+        experiment = Experiment(id=1, name='CongoSameDiff', slug='congosamediff', rounds=4)
+        actions = congo_same_diff.first_round(experiment)
+        assert len(actions) >= 1
+        assert isinstance(actions[0], PlaylistAction)
+        assert isinstance(actions[1], Explainer)
+
+    def test_next_round_final_round(self):
+        congo_same_diff, session = create_context('final')
+
+        session.get_next_round = lambda: 6
+        final_action = congo_same_diff.next_round(session)
+
+        assert isinstance(final_action, Final)
+
+    def test_next_round_practice_trial(self):
+        congo_same_diff, session = create_context('practice')
+
+        session.get_next_round = lambda: 2
+        practice_action = congo_same_diff.next_round(session)
+
+        assert isinstance(practice_action, Trial)
+        assert practice_action.feedback_form.form[0].question == 'Is the third sound the SAME or DIFFERENT as the first two sounds?'
+
+    def test_next_round_non_practice_trial(self):
+        congo_same_diff, session = create_context('non-practice')
+
+        session.get_next_round = lambda: 4
+        non_practice_action = congo_same_diff.next_round(session)
+
+        assert isinstance(non_practice_action, Trial)
+        assert non_practice_action.feedback_form.form[0].question == 'Is the third sound the SAME or DIFFERENT as the first two sounds?'
+
+    def test_get_next_trial(self):
+        congo_same_diff, session = create_context('next_trial')
+        subset = session.playlist.section_set.exclude(tag__contains='practice')
+        trial_index = 1
+        is_practice = True
+        trial_action = congo_same_diff.get_next_trial(session, subset, trial_index, is_practice)
+        assert isinstance(trial_action, Trial)
+        assert trial_action.feedback_form.form[0].question == 'Is the third sound the SAME or DIFFERENT as the first two sounds?'
+
+    def test_get_final_round(self):
+        congo_same_diff, session = create_context('final')
+
+        final_action = congo_same_diff.get_final_round(session)
+
+        assert isinstance(final_action, Final)
+        assert final_action.final_text == 'Thank you for participating!'
+
+
+def create_context(name):
+    playlist = PlaylistModel.objects.create(name=f'CongoSameDiff{name}')
```

Comment:
Why create different playlist and experiment objects every time? I cannot see why the same playlist & experiment couldn't be reused over all tests.

Reply:
I think I had some problems with them interfering with each other. Don't remember exactly but I will check & see what happens if I put it back into the setupTestData method

Reply:
Ah, that could also be solved by using Django's `setUp` method instead of the (class) `setUpTestData` method, see [documentation](https://docs.djangoproject.com/en/5.0/topics/testing/overview/#writing-tests). I forgot why we used the `setUpTestData` method in the first place, perhaps to speed up tests. `setUp` should destroy the objects after every test, and reconstruct them for the next.

Reply:
In the end I didn't really need the `create_context` method and it worked with just using the `setupTestData` method.